### ISSUE TITLE
GZip deflate causing JSON parse error when using gem

### ISSUE THREAD

**Issue by KNejad** [campaignmonitor/createsend-ruby#58]
## To reproduce

1. Have a list containing 8 SmartMail templates (this is over 1KB of content for us)
2. Call this method

```
      CreateSend::Transactional::SmartEmail.list(
        { api_key: Rails.application.secrets.campaignmonitor_api_key },
        clientID: Rails.application.secrets.campaignmonitor_client_id
      )
```

This crashes with 

```
JSON::ParserError: 765: unexpected token at ''
…d/bundle/ruby/2.3.0/gems/json-2.1.0/lib/json/common.rb:  156:in `parse'
…d/bundle/ruby/2.3.0/gems/json-2.1.0/lib/json/common.rb:  156:in `parse'
…ruby/2.3.0/gems/httparty-0.15.6/lib/httparty/parser.rb:  123:in `json'
…ruby/2.3.0/gems/httparty-0.15.6/lib/httparty/parser.rb:  143:in `parse_supported_format'
…ruby/2.3.0/gems/httparty-0.15.6/lib/httparty/parser.rb:  108:in `parse'
…ruby/2.3.0/gems/httparty-0.15.6/lib/httparty/parser.rb:   67:in `call'
…uby/2.3.0/gems/httparty-0.15.6/lib/httparty/request.rb:  382:in `parse_response'
…uby/2.3.0/gems/httparty-0.15.6/lib/httparty/request.rb:  350:in `block in handle_response'
…by/2.3.0/gems/httparty-0.15.6/lib/httparty/response.rb:   25:in `parsed_response'
…by/2.3.0/gems/httparty-0.15.6/lib/httparty/response.rb:   88:in `method_missing'
…send-4.1.1/lib/createsend/transactional_smart_email.rb:    9:in `list'
…1109093951/app/controllers/my_controller.rb:   63:in `block in get_campaign_monitor_autoresponders_list'
```

We have investigated this and the below HTTParty call is what the gem is roughly doing based on this line: 
https://github.com/campaignmonitor/createsend-ruby/blob/master/lib/createsend/createsend.rb#L220
And the line that sets the header is: https://github.com/campaignmonitor/createsend-ruby/blob/master/lib/createsend/createsend.rb#L118
```ruby
r = HTTParty.get("https://api.createsend.com/api/v3.1/transactional/smartemail", {headers: { "Accept-Encoding" => "gzip, deflate"}, query: options, basic_auth: { username: Rails.application.secrets.campaignmonitor_api_key, password: 'x'}})
r.body => #Still GZipped content. JSON cannot parse this!
```
Expected:
```ruby
r = HTTParty.get("https://api.createsend.com/api/v3.1/transactional/smartemail", {query: options, basic_auth: { username: Rails.application.secrets.campaignmonitor_api_key, password: 'x'}})
r.body => # String encoded JSON that parses just fine!
```

**Comment by cmtiml** [comment]
Hello @KNejad,

We have merged the change for removing "Accept-Encoding"=>"gzip, deflate" and released it now for version 4.1.2 (https://rubygems.org/gems/createsend).

Please kindly check and verify if it solved the issue.
Thank you very much for your solution to the issue.

**Comment by KNejad** [comment]
Hello @cmtiml Yes this solved the issue we had. Thank you! :)
